pkg/snapshot/snapshot_writer.go

```diff
@@ -19,7 +19,7 @@ import (
 // Writer creates a snapshot using an artifact path as location for the snapshot
 // data.
 type Writer interface {
-	Write(ctx context.Context, owner v1alpha1.MutationObject, sourceDir string, identity ocmmetav1.Identity) (string, error)
+	Write(ctx context.Context, owner client.Object, snapshotName, sourceDir string, identity ocmmetav1.Identity) (string, error)
```

Comment:
Could we use the `[SnapshotProducer](https://github.com/open-component-model/ocm-controller/blob/034e1df8d4f384259b37adcf4c8cc5cf5f9554fe/api/v1alpha1/snapshot_types.go#L29) interface here? 

Anything that write a snapshot should satisfy this interface.

Reply:
I was thinking about this, but that's not actually completely true. The product controller writes a snapshot that's right, but it doesn't do anything with it and might actually write several depending on the case what is getting synced and doesn't want to save them in its status at all. So I'm not sure here. That said, I could implement the interface and provide no value or provide a value by calling generating the random name. WDYT?

Reply:
Maybe we should rename the interface SnapshotWriter? 

Would it really write multiple snapshots or a sequence of snapshot versions?

Reply:
Hmm, hmmm I guess one productgenerator writes only a single snapshot. 🤔 The one that it wants the git sync to sync.

Reply:
hopefully. that's all 😄

Reply:
Can't use `SnapshotProducer` because that's not a client.Object so it can't be an `owner`./

Reply:
I can embed `client.Object` but that's weird from the MutationObject's perspective that already DOES include it.